### PR TITLE
batches: Add Stories for SSBC Components

### LINKED ISSUE #38658: Increase storybook coverage of SSBC components
We've quickly added a lot of components of which a lot are not currently covered by a story. It'd be very nice if every component we have would be covered by a storybook again.

### PR DESCRIPTION
**Pull request by Piszmog**
Closes #38658.

Simply add additional stories.

Many larger/complex components are already covered by parent/ancestor component stories. So added stories for the smaller components and used Storybook Knobs to be able to view many different scenarios without writing a bunch of cluttery stories that are technically already covered.

## Test plan

Add new Stories

## App preview:

- [Web](https://sg-web-rc-ssbc-storybook.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-lafttjdsnb.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]

**Review by LawnGnome — APPROVED** [review]
Thanks for taking this on!

File: client/web/src/enterprise/batches/batch-spec/execute/workspaces/StepStateIcon.story.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    'Not Started': { startedAt: null },
+    Running: { startedAt: 'start-time' },
+    Success: { startedAt: 'start-time', finishedAt: 'start-time', exitCode: 0 },
+    Failed: { startedAt: 'start-time', finishedAt: 'start-time', exitCode: 1 },
+}
+
+export const StepStateIconStory: Story = () => (
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    <WebStory>{props => <StepStateIcon step={select('State', options, options.Success)} {...props} />}</WebStory>
```
Comment (courier-new) [inline client/web/src/enterprise/batches/batch-spec/execute/workspaces/StepStateIcon.story.tsx:29]: Maybe same suggestion for this one? Maybe the story could even label each icon, so it's extra useful as a visual reference for design?

File: client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceStateIcon.story.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    title: 'web/batches/batch-spec/execute/workspaces/WorkspaceStateIcon',
+    decorators: [decorator, withKnobs],
+}
+
+export default config
+
+export const WorkspaceStateIconStory: Story = () => (
+    <WebStory>
+        {props => (
+            <WorkspaceStateIcon
```
Comment (courier-new) [inline client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceStateIcon.story.tsx:22]: You can probably guess my feedback here. 🙂

File: client/web/src/enterprise/batches/batch-spec/execute/workspaces/Workspaces.story.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        result: { data: mockWorkspaces(50) },
+        nMatches: Number.POSITIVE_INFINITY,
+    },
+])
+
+export const WorkspacesStory: Story = () => (
+    <WebStory>
+        {props => (
+            <MockedStoryProvider link={MOCKS}>
+                <Workspaces batchSpecID="1" selectedNode="workspace1" executionURL="" {...props} />
```
Comment (courier-new) [inline client/web/src/enterprise/batches/batch-spec/execute/workspaces/Workspaces.story.tsx:37]: Great idea to have a selected node on this one! We've gotten bitten by regressions on that at least twice already. 😂

**Review by courier-new — APPROVED** [review]
Awesome, thanks so much Randell! I like your approach leaning into the knobs on these! I had a couple small suggestions but otherwise looks great.

I'm also wondering if you came out of this with any strong feelings about which of these stories, if any, it would be useful to turn Chromatic snapshot comparisons in CI back on for?

**Comment by Piszmog** [comment]
I think turning Chromatic comparisons on for some would be helpful. Though, no too strong of a feeling. Many of the components I added stories for are already covered under a much larger story. But I think have comparisons at a granular scale would cause less headaches.

**Pull request merged**